lambda/main.tf

```diff
@@ -0,0 +1,94 @@
+resource "aws_lambda_function" "lambda_function" {
+  function_name = var.function_name
+  handler       = var.handler
+  role          = aws_iam_role.lambda_iam_role.arn
+  runtime       = var.runtime
+  filename      = startswith(var.runtime, "java") ? "${path.module}/functions/generic.jar" : "${path.module}/functions/generic.zip"
+  timeout       = var.timeout_seconds
+  memory_size   = var.memory_size
+
+  ephemeral_storage {
+    size = var.storage_size
+  }
+
+  reserved_concurrent_executions = var.reserved_concurrency
+  tags                           = var.tags
+  environment {
+    variables = local.all_env_vars
+  }
+
+  dynamic "file_system_config" {
+    for_each = var.efs_access_points
+    content {
+      arn              = file_system_config.value.access_point_arn
+      local_mount_path = file_system_config.value.mount_path
+    }
+  }
+  dynamic "vpc_config" {
+    for_each = var.vpc_config
+    content {
+      subnet_ids         = vpc_config.value.subnet_ids
+      security_group_ids = vpc_config.value.security_group_ids
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [filename]
+  }
+}
+
+resource "aws_cloudwatch_log_group" "lambda_log_group" {
+  name = "/aws/lambda/${aws_lambda_function.lambda_function.function_name}"
+  tags = var.tags
+}
```

Comment:
Add and use a new variable to expire logs?

Reply:
I've added a log_retention variable and defaulted it to 30 days.